managed/services/telemetry/telemetry.go

```diff
@@ -208,17 +210,29 @@ func (s *Service) processSendCh(ctx context.Context) {
 }
 
 func (s *Service) prepareReport(ctx context.Context) *pmmv1.ServerMetric {
+	initializedDataSources := make(map[string]DataSource)
 	telemetryMetric, _ := s.makeMetric(ctx)
-
 	var totalTime time.Duration
-	for _, telemetry := range s.config.telemetry {
-		// locate DS
-		ds, err := s.LocateTelemetryDataSource(telemetry.Source)
+
+	// initialize datasources
+	for sourceName, dataSource := range s.dataSourcesMap {
+		err := dataSource.Init(ctx)
 		if err != nil {
-			s.l.Debugf("failed to lookup telemetry datasource for [%s]:[%s]", telemetry.Source, telemetry.ID)
+			s.l.Error("Telemetry datasource init failed", err)
```

Comment:
Better to use `warn` level for cases when work can be continued.
```suggestion
			s.l.Warnf("Telemetry datasource %s init failed: %v", sourceName, err)
```

Reply:
Changed.